noir_by_example/README.md

```diff
@@ -0,0 +1,66 @@
+# Examples showcasing syntax, libraries, and data structures
+
+## General
+
+These will show Noir side-by-side against [Rust by example](https://doc.rust-lang.org/rust-by-example/).
+
+The goal is to highlight the similarities and differences in the language and design patterns.
+
+### Tooling
+
+You will need:
+
+- `nargo` installed [via noirup](https://noir-lang.org/docs/getting_started/installation/). Uses 0.34.0 (`noirup -v 0.34.0`)
+- `cargo` installed [via rustup](https://www.rust-lang.org/tools/install) (optional: to see Rust output)
+
+### What is not covered
+
+The surrounding build artifacts (and additional proving/verifying artifacts for Noir) are not explored in this example.
+
+## Usage
+
+The following commands can be run at the top level `noir_by_example`, or withing the specific topic directory.
+Eg: `cd simple_macros`
+
+### Compiling
+
+- `nargo compile`
+- `cargo build`
+
+### Running main
+
+- Noir:
+  - `nargo check` # May need to populate values in Prover.toml for topics that need it
+  - `nargo execute`
+- Rust:
+  - `cargo run` # topic-level only
```

Comment:
Sorry, missed this error on the last pass.

![image](https://github.com/user-attachments/assets/02326053-f1d2-4301-94aa-dedf3781a026)

Reply:
Showcasing the difference of Noir executing all binaries from top-level Nargo.toml (even if called from topic-level directory). On the other hand `cargo run` must be called from only the topic-level (as per comment). It does not ascend to the top level Cargo.toml file, and trying to run this file fails because I've not specified a single binary to run.
I could specify something in cargo that runs each topic binary, but there are probably enough differences to track at this stage. Will make PR if someone trips on it.